### PR TITLE
kubectl-plan: try to pretty-print diffs

### PR DESCRIPTION
**Pull request by justinsb**
- **feat: include more diff details in kubectl-plan output**
- **tests: golden-test "pretty" output from kubectl-plan also**
- **[kubectl-plan] tests: pre-create namespace, trim dynamic fields**

### CODE REVIEW

File: experiments/kubectl-plan/ssh-auth-sock.socket
Comment (yuwenma) [inline experiments/kubectl-plan/ssh-auth-sock.socket:1]: remove?

File: experiments/kubectl-plan/pkg/plan/testdata/simpleupdate/_plan.yaml [hunk 1]
```diff
@@ -4,6 +4,19 @@ spec:
   actions:
   - action: ApplyChanges
     apiVersion: v1
+    diff:
```
Comment (yuwenma) [inline experiments/kubectl-plan/pkg/plan/testdata/simpleupdate/_plan.yaml:7]: Would like to see how we do diff for slice and map

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve
/hold for ssh socket
Really cool!

File: experiments/kubectl-plan/ssh-auth-sock.socket
Comment (justinsb) [reply experiments/kubectl-plan/ssh-auth-sock.socket:1]: Good call, thank you!

**Pull request merged**